core/src/nats.rs

```diff
@@ -300,24 +296,46 @@ impl TypedNats {
 
         let consumer = stream
             .create_consumer(async_nats::jetstream::consumer::pull::Config {
-                deliver_policy: DeliverPolicy::Last,
+                deliver_policy,
                 filter_subject: subject.subject.clone(),
                 ..async_nats::jetstream::consumer::pull::Config::default()
             })
             .await
             .to_anyhow()?;
 
-        let result = match timeout(
-            Duration::from_secs(1),
-            consumer.stream().messages().await.to_anyhow()?.next(),
-        )
-        .await
-        {
-            Ok(Some(v)) => v.to_anyhow()?,
-            _ => return Ok(None),
-        };
-
-        Ok(Some(serde_json::from_slice(&result.payload)?))
+        let mut result: Vec<T> = Vec::new();
+
+        loop {
+            let mut messages = consumer.fetch().messages().await.to_anyhow()?;
+            let mut done = true;
+
+            while let Some(v) = messages.next().await {
+                let v = v.to_anyhow()?;
+                done = false;
+
+                result.push(serde_json::from_slice(&v.payload)?);
+
+                if v.info().to_anyhow()?.pending == 0 {
+                    done = true;
+                    break
+                }                
+            }
+
+            if done {
+                break;
+            }
+        }
+
+        Ok(result)
+    }
+
+    pub async fn get_latest<T>(&self, subject: &SubscribeSubject<T>) -> Result<Option<T>>
+    where
+        T: TypedMessage<Response = NoReply> + JetStreamable,
+    {
+        let mut result = self.get_all(subject, DeliverPolicy::Last).await?;
```

Comment:
with deliverpolicy::last can we just call consumer.messages().next()?

Reply:
Hmm good call, I guess we could.

Reply:
Turns out this isn't used anywhere and I have doubts about whether it's useful as-is, so I've just removed it.